src/code/z_parameter.c

```diff
@@ -2385,29 +2410,32 @@ void Interface_UpdateMagicBar(GlobalContext* globalCtx) {
     s16 borderChangeR;
     s16 borderChangeG;
     s16 borderChangeB;
-    s16 temp;
-
-    switch (gSaveContext.unk_13F0) {
-        case 8:
-            temp = gSaveContext.magicLevel * 0x30;
-            if (gSaveContext.unk_13F4 != temp) {
-                if (gSaveContext.unk_13F4 < temp) {
-                    gSaveContext.unk_13F4 += 8;
-                    if (gSaveContext.unk_13F4 > temp) {
-                        gSaveContext.unk_13F4 = temp;
+    s16 temp; // magicCapacity or magicBorderIndex
+
+    switch (gSaveContext.magicState) {
+        case MAGIC_STATE_GROW_METER:
+            // Step magicCapacityDrawn to what is magicCapacity
```

Comment:
As here: it's basically the same as MAGIC_STATE_FILL, but for the bar itself, right?

Reply:
In a sense. The way this block is designed is that it just moves the bar drawn to where the true capacity is. In theory, it could be moved wider or smaller (i.e. it doesn't specifically have to grow), it's just that in practice it only grows based on how this case is used

Reply:
So it could be `UPDATE_WIDTH`, even.

Reply:
`WIDTH` could apply to a lot of things so I think it'd be important to specify bar in there, but yeah I'd be fine with that. Could maybe even be `SYNC_BAR_WIDTH` in the sense that it "sync"s the `magicCapacityDrawn` to the true `magicCapacity` just as an idea